Chocolate_Vending_Machine.java

```diff
@@ -0,0 +1,240 @@
+		intialize_Item_Inventory();
+	}
+
+	public void displayMenu() {
+		System.out.println("Welcome");
+		System.out.println("MENU:");
+		System.out.println("1. Dairy Milk				Rs. 10/-");
+		System.out.println("2. Munch				Rs. 5/-");
+		System.out.println("3. Milky Bar				Rs. 15/-");
+		System.out.println("4. 5 Star				Rs. 5/-");
+		System.out.println(
+				"Please press the corresponding no. from the menu on the keypad and then quantity to get your chocolate ");
+
+	}
+
+	public boolean hasSufficientItem(int s_no, int qty) {
+		Item items_in[] = Item.values();
+		int avail_qty = items.get(items_in[s_no - 1]);
+		if (avail_qty < qty||qty<0) {
+			return false;
+		}
+		return true;
+	}
+
+	public boolean sufficientMoneyGiven(ArrayList<Integer> money, int s_no, int qty) throws NoSufficientMoneyEnteredByCustomer {
+		int tot_Money = 0;
+		for (int i = 0; i < money.size() - 1; i++) {
+			tot_Money += money.get(i);
+		}
+		Item items_in[] = Item.values();
+		int actualPrice = items_in[s_no - 1].getPrice(items_in[s_no - 1]) * qty;
+		if (tot_Money < actualPrice) {
+			throw new NoSufficientMoneyEnteredByCustomer("You have not entered sufficient money to place order!! Please start again");
+		}
+		return true;
+	}
+
+	public ArrayList<Coin> returnChange(ArrayList<Integer> money, int s_no, int qty)
+			throws NoSufficientChangeException {
+		ArrayList<Coin> changes = new ArrayList<>();
+		int coindeduct[] = new int[4];
+		int tot_Money = 0;
+		for (int i = 0; i < money.size() - 1; i++) {
+			tot_Money += money.get(i);
+			switch (money.get(i)) {
+			case 10:
+				addCoin(Coin.ten, 1);
+				break;
+			case 5:
+				addCoin(Coin.five, 1);
+				break;
+			case 2:
+				addCoin(Coin.two, 1);
+				break;
+			case 1:
+				addCoin(Coin.one, 1);
+				break;
+			}
+		}
+		Item items_in[] = Item.values();
+		int actualPrice = items_in[s_no - 1].getPrice(items_in[s_no - 1]) * qty;
+		int diffMoney = tot_Money - actualPrice;
+
+		if (diffMoney > 0) {
+			while (diffMoney > 0) {
+				if (diffMoney >= Coin.ten.getDenomination() && cash.get(Coin.ten) >= coindeduct[0]) {
+					diffMoney -= Coin.ten.getDenomination();
+					coindeduct[0]++;
+					changes.add(Coin.ten);
+				} else if (diffMoney >= Coin.five.getDenomination() && cash.get(Coin.five) >= coindeduct[1]) {
+					diffMoney -= Coin.five.getDenomination();
+					coindeduct[1]++;
+					changes.add(Coin.five);
+				} else if (diffMoney >= Coin.two.getDenomination() && cash.get(Coin.two) >= coindeduct[2]) {
+					diffMoney -= Coin.two.getDenomination();
+					coindeduct[2]++;
+					changes.add(Coin.two);
+				} else if (diffMoney >= Coin.one.getDenomination() && cash.get(Coin.two) >= coindeduct[3]) {
+					diffMoney -= Coin.one.getDenomination();
+					coindeduct[3]++;
+					changes.add(Coin.one);
+				} else {
+					throw new NoSufficientChangeException("Machine does not have sufficient change");
+				}
+			}
+		}
+		return changes;
+	}
+
+	private void updateCashInventory(ArrayList<Coin> changes) {

```

Comment:
These two functions have confusing names.  Since they only ever remove items, I would call that out with the names